File: src/main/java/dev/tuhkanens/comfortlib/database/base/provider/SQLiteBase.java
package dev.tuhkanens.comfortlib.database.base.provider;

import dev.tuhkanens.comfortlib.ComfortAPI;
import dev.tuhkanens.comfortlib.database.base.DatabaseBase;
import org.jdbi.v3.core.Jdbi;

import java.nio.file.Path;

public class SQLiteBase extends DatabaseBase {

    private final Path directory;

    public SQLiteBase(Path directory) {
        this.directory = directory;
    }

    @Override
    protected Jdbi createConnection() {
        String databaseName = ComfortAPI.CONFIG
                .getNode()
                .node("database", "database")
                .getString("comfort");

        Path databasePath = directory.resolve(databaseName + ".db");
        return Jdbi.create("jdbc:sqlite:" + databasePath);
    }
}

File: src/main/kotlin/dev/tuhkanens/comfortlib/database/base/SQLiteBase.kt
package dev.tuhkanens.comfortlib.database.base

import dev.tuhkanens.comfortlib.ComfortAPI
import dev.tuhkanens.comfortlib.api.ConfigAPI
import dev.tuhkanens.comfortlib.database.DatabaseBase
import org.jetbrains.exposed.v1.jdbc.Database
import java.nio.file.Path

class SQLiteBase(private val directory: Path) : DatabaseBase() {
    override fun createConnection(): Database {
        val databaseName = ComfortAPI.getAPI<ConfigAPI>().getNode().node("database", "database").getString("comfort")
        val path = directory.resolve("$databaseName.db").toString()
        return Database.connect("jdbc:sqlite:$path", driver = "org.sqlite.JDBC")
    }
}